File: 0136-single-number/0136-single-number.py
"""
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        occurs = {}
        for n in nums:
            if occurs.get(n) == None:
                occurs[n] = 1
            else:   
                occurs[n] += 1
        
        for k, v in occurs.items():
            if v == 1:
                return k
            
        return None
"""
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        res = 0
        for n in nums:
            res ^= n
            
        return res    

File: 0136-single-number/0136-single-number.go
/*
func singleNumber(nums []int) int {
    occ := map[int]int{}
    for _, n := range nums {
        occ[n] += 1
    }
    for k, v := range occ {
        if v == 1 {
            return k
        }
    }
    return 0
}
*/

func singleNumber(nums []int) int {
    res := 0
    for _, n := range nums {
        res ^= n
    }
    return res
}